### PR TITLE
Use workflow short name rather than long name, state migrate

### PR DESCRIPTION
**Pull request by slevenick**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8832

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
workflows: fixed a bug in `google_workflows_workflow` that could cause inconsistent final plan errors when using the `name` field in other resources
```

### CODE REVIEW

File: mmv1/products/workflows/terraform.yaml [hunk 1]
```diff
@@ -23,8 +23,9 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       update_minutes: 6
       delete_minutes: 6
     autogen_async: true
-    id_format: '{{name}}'
+    id_format: 'projects/{{project}}/locations/{{region}}/workflows/{{name}}'
     exclude_import: true
+    schema_version: 1
```
Comment (rileykarson) [inline mmv1/products/workflows/terraform.yaml:28]: drive-by q: do we need a state upgrader, or can we just update the value in state on next refresh? If we can't refresh, how come?

Comment (slevenick) [reply mmv1/products/workflows/terraform.yaml:28]: Name is used to call the read function, if name has been set to a long-form it won't correctly call I believe?

Comment (rileykarson) [reply mmv1/products/workflows/terraform.yaml:28]: Got it, thanks!

File: mmv1/third_party/terraform/tests/resource_workflows_workflow_test.go [hunk 2]
```diff
@@ -68,40 +69,85 @@ EOF
...
+			if err != nil {
+				t.Error(err)
+			}
+
+			for k, v := range tc.Expected {
+				if actual[k] != v {
+					t.Errorf("expected: %#v -> %#v\n got: %#v -> %#v\n in: %#v",
+						k, v, k, actual[k], actual)
+				}
+			}
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_workflows_workflow_test.go:150]:
```suggestion
			for _, expectedName := range tc.Expected {
				if actual["name"] != expectedName {
					t.Errorf("expected: name -> %#v\n got: name -> %#v\n in: %#v",
						expectedName, actual["name"], actual)
				}
			}
```

readability nit

**Review by ScottSuarez — APPROVED** [review]
